services/galley/src/Galley/API/Update.hs

```diff
@@ -892,16 +892,16 @@ addToConversation (bots, lothers) rothers (usr, usrRole) conn locals remotes c =
   now <- liftIO getCurrentTime
   localDomain <- viewFederationDomain
   (e, lmm, rmm) <- Data.addMembersWithRole localDomain now (Data.convId c) (usr, usrRole) mems
-  let others = catMembers localDomain lothers rothers
-      mm = catMembers localDomain lmm rmm
+  let mm = catMembers localDomain lmm rmm
       qcnv = Qualified (Data.convId c) localDomain
       qusr = Qualified usr localDomain
   -- FUTUREWORK: parallelise federated requests
   traverse_ (uncurry (updateRemoteConversations now mm qusr qcnv))
     . Map.assocs
     . partitionQualified
-    . map fst
-    $ others
+    . nubOrd
+    . map (unTagged . rmId)
+    $ rmm <> rothers
```

Comment:
If I am getting this right, this traversing rightfully notifies remote users. Is there an equivalent for local users?

Reply:
This doesn't directly notify users, but it calls the `update-conversation-memberships` federation RPC, which will in turn notify users on its backend.

Notifications for local users are sent in the two statements that follow.